### PR TITLE
feat(ARCO-279): add custom zmq listener

### PR DESCRIPTION
**Pull request by pawellewandowski98**
## Description of Changes

The code related to the zmq connection from the `go-bitcoin` library has been refactored and moved to our repository

## Linked Issues / Tickets

ARCO-279

## Testing Procedure

Describe the tests you've added or any testing steps you've taken.

- [ ] I have added new unit tests
- [ ] All tests pass locally
- [ ] I have tested manually in my local environment

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have updated `CHANGELOG.md` with my changes

### CODE REVIEW

File: cmd/arc/services/metamorph.go [hunk 1]
```diff
@@ -201,19 +199,10 @@ func StartMetamorph(logger *slog.Logger, arcConfig *config.ArcConfig, cacheStore
 			continue
 		}
 
-		zmq := metamorph.NewZMQ(zmqURL, statusMessageCh, logger)
+		zmq := metamorph.NewZMQ(context.Background(), zmqURL, statusMessageCh, nil, logger)
```
Comment (boecklim) [inline cmd/arc/services/metamorph.go:202]: IMO we should create zmq handler here and pass it
```suggestion
zmqHandler := NewZMQHandler(ctx, zmqURL, logger)
zmq := metamorph.NewZMQ(context.Background(), zmqURL, statusMessageCh, zmqHandler, logger)
```

File: internal/metamorph/zmq.go [hunk 2]
```diff
@@ -56,21 +77,26 @@ type CollidingTx struct {
...
 }
 
-func NewZMQ(zmqURL *url.URL, statusMessageCh chan<- *PeerTxMessage, logger *slog.Logger) *ZMQ {
+type ZMQI interface {
+	Subscribe(string, chan []string) error
+}
+
+func NewZMQ(ctx context.Context, zmqURL *url.URL, statusMessageCh chan<- *PeerTxMessage, zmqHandler ZMQI, logger *slog.Logger) *ZMQ {
+	if zmqHandler == nil {
+		zmqHandler = NewZMQHandler(ctx, zmqURL, logger)
```
Comment (boecklim) [inline internal/metamorph/zmq.go:86]: If the zmq handler is `nil`, I think we should return an error here

File: cmd/arc/services/metamorph.go (hunk 1 above)
Comment (pawellewandowski98) [reply cmd/arc/services/metamorph.go]: Done

File: internal/metamorph/zmq.go (hunk 2 above)
Comment (pawellewandowski98) [reply internal/metamorph/zmq.go]: Done

**Review by boecklim — APPROVED** [review]

**Pull request merged**